libs/community/langchain_community/tools/azure_cognitive_services/content_safety.py

```diff
@@ -0,0 +1,151 @@
+        "A wrapper around Azure AI Content Safety. "
+        "Useful for when you need to identify the sentiment of text and whether"
+        " or not a text is harmful."
+        "Input must be text (str)."
+    )
+
+    def __init__(
+        self,
+        *,
+        content_safety_key: Optional[str] = None,
+        content_safety_endpoint: Optional[str] = None,
+    ) -> None:
+        """
+        Initialize the AzureContentSafetyTextTool with the given API key and endpoint.
+
+        If not provided, the API key and endpoint are fetched from environment
+        variables.
+
+        Args:
+            content_safety_key (Optional[str]):
+                The API key for Azure Content Safety API. If not provided, it will
+                be fetched from the environment variable 'CONTENT_SAFETY_API_KEY'.
+            content_safety_endpoint (Optional[str]):
+                The endpoint URL for Azure Content Safety API. If not provided, it
+                will be fetched from the environment variable
+                'CONTENT_SAFETY_ENDPOINT'.
+
+        Raises:
+            ImportError: If the 'azure-ai-contentsafety' package is not installed.
+            ValueError: If API key or endpoint is not provided and environment
+                variables are missing.
+        """
+        content_safety_key = content_safety_key or os.environ["CONTENT_SAFETY_API_KEY"]
+        content_safety_endpoint = (
+            content_safety_endpoint or os.environ["CONTENT_SAFETY_ENDPOINT"]
+        )
+        try:
+            import azure.ai.contentsafety as sdk
+            from azure.core.credentials import AzureKeyCredential
+
+            content_safety_client = sdk.ContentSafetyClient(
+                endpoint=content_safety_endpoint,
+                credential=AzureKeyCredential(content_safety_key),
+            )
+
+        except ImportError:
+            raise ImportError(
+                "azure-ai-contentsafety is not installed. "
+                "Run `pip install azure-ai-contentsafety` to install."
+            )
+        super().__init__(
+            content_safety_key=content_safety_key,
+            content_safety_endpoint=content_safety_endpoint,
+            content_safety_client=content_safety_client,
+        )
+
+    def _sentiment_analysis(self, text: str) -> Dict:
+        """
+        Perform sentiment analysis on the provided text.
+
+        This method uses the Azure Content Safety Client to analyze the text and
+        determine its sentiment and safety categories.
+
+        Args:
+            text (str): The text to be analyzed.
+
+        Returns:
+            Dict: The analysis results containing sentiment and safety categories.
+        """
+        from azure.ai.contentsafety.models import AnalyzeTextOptions
+
+        request = AnalyzeTextOptions(text=text)
+        response = self.content_safety_client.analyze_text(request)
+        result = response.categories_analysis
+        return result
```

Comment:
According to the signature, this should return a dict. I'm not sure if that's in the BaseTool contract, but it does seem like the other tools convert this to a built-in type. I believe the `result` here to be a list[TextCategoriesAnalysis]. I think we can transform this into a dict by having the category as the key and the severity as the value:

For example:

```python
result_dict: dict[str, int] = {
    "Hate": 2,
    "SelfHarm": 0,
    "Sexual": 0,
    "Violence": 1
}
```

Reply:
This was a mistake of mine. However when looking over the other tools they seem to prefer returning a string over a list or dict. I've changed this tool to return in the format you mentioned:

```python
    '''Hate: 2\n
    SelfHarm: 0\n
    Sexual: 0\n
    Violence: 1\n'''
```